doc/pattern_tools/gen_pattern.py

```diff
@@ -140,6 +148,69 @@ def make_radon_checkerboard_pattern(self):
                           cy=(y * spacing) + y_spacing + r, r=r, fill=color, stroke="none")
                 self.g.append(dot)
 
+    @staticmethod
+    def _create_marker_bits(markerSize_bits, byteList):
+        
+        marker = np.zeros((markerSize_bits+2, markerSize_bits+2))
+        bits = marker[1:markerSize_bits+1, 1:markerSize_bits+1]
+            
+        for i in range(markerSize_bits):
+            for j in range(markerSize_bits):
+                bits[i][j] = int(byteList[i*markerSize_bits+j])
+     
+        return marker
+
+    def make_charuco_board(self):
+        if (self.aruco_marker_size>self.square_size):
+            print("Error: Aruco marker can not be more than square size!")
+            return
+
+        if (self.dict_file.split(".")[-1] == "gz"):
+            fin = gzip.open(self.dict_file, 'r')
```

Comment:
I recommend to use `with .. as..:` notation here and bellow for not compressed files, e.g.:
```
with gzip.open(self.dict_file, 'r') as fin:
            json_bytes = fin.read()
            json_str = json_bytes.decode('utf-8')
            dictionary = json.loads(json_str)
```